### PR TITLE
Documentation: Adding a docs page explaining the deprecated blocks API

### PR DESCRIPTION
**Pull request by youknowriad**

### CODE REVIEW

File: docs/deprecated-blocks.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Deprecated Blocks
+
+When updating static blocks markup and attributes, block authors need to consider existing posts using the old versions of their block. In order to provide a good upgrade path, you can choose one of the following strategies:
+
+ - Do not deprecate the block and create a new one (a different name)
+ - Provide a "deprecated" version of the block allowing users opening these blocks in Gutenberg to edit them using the updated block.
+
+A block can have several deprecated versions, Gutenberg will try them one after another until finding the one that matches the saved markup.
```
Comment (aduth) [inline docs/deprecated-blocks.md]: Grammar: Probably more appropriate to use a period (".") in place of the comma here.

File: docs/deprecated-blocks.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+To declare a "deprecated version", you need to copy the old `attributes`, `support` and `save` properties from the previous old definition to the `deprecated` property of the udpated block.
```
Comment (aduth) [inline docs/deprecated-blocks.md]: Don't think we need the quotes here on "deprecated version".

File: docs/deprecated-blocks.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	// ... other block properties go here
+
+	attributes: {
+		text: {
+			type: 'string',
+			default: 'some random value',
+		}
+	},
+
+	save: function( props ) {
```
Comment (aduth) [inline docs/deprecated-blocks.md]: For ESNext, maybe `save( props ) {`

File: docs/deprecated-blocks.md (hunk 2 above)
Comment (aduth) [reply docs/deprecated-blocks.md]: "previous old" is redundant, we can choose one or the other.

**Review by aduth — APPROVED** [review]

File: docs/deprecated-blocks.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+{% ES5 %}
+```js
+var el = wp.element.createElement,
+	registerBlockType = wp.blocks.registerBlockType;
+
+registerBlockType( 'gutenberg/block-with-deprecated-version', {
+
+	// ... other block properties go here
+
+	attributes: {
```
Comment (mtias) [inline docs/deprecated-blocks.md]: Should we have two examples, one using the same set of attributes even with new markup?

Comment (youknowriad) [reply docs/deprecated-blocks.md]: Sure, I added an example with `migrate` as well

**Pull request merged**